### ISSUE TITLE
[DataGrid] Allow sortable column header cells to be navigated to and triggered via keyboard

### ISSUE THREAD

**Issue by msteitle** [mui-org/material-ui-x#1022]
Make DataGrid column headers focusable via keyboard <tab> navigation, and allow sort state to be triggered by <space> and <enter>. For accessibility.

- [x] I have searched the [issues](https://github.com/mui-org/material-ui-x/issues) of this repository and believe that this is not a duplicate.

## Summary 💡

DataGrid table body cells can be fully navigated via the keyboard. There is no way to navigate to column header cells in order to toggle sort state on sortable columns. These cells should be focusable, as they act like buttons for sorting. The sort direction indicator icon is keyboard accessible, but it does not become visible until sort has been applied. Additionally, when the sort state returns to natural (no sort applied) the icon disappears and focus is lost.

A user should be able to navigate to sortable column header cells with the <tab> key & <shift>+<tab>, as the web standard for navigating between focusable elements.

Once focused, sort state should be triggered on the focused column header cell with either <space> or <enter> keypress.

## Examples 🌈

https://webaim.org/techniques/keyboard/#intro

Screen cap of focusing header cell and triggering sort via keyboard on example at https://www.w3.org/TR/wai-aria-practices/examples/grid/dataGrids.html
https://imgur.com/a/YtnwyZs

## Motivation 🔦

We are trying to accomplish accessibility compliance.
The lack of this feature is problematic because we cannot remain accessibility compliant, preventing some of our users from being able to interact with the grid. There is no way to correct the behavior via the DataGrid API. Additionally, if the prop `tabindex="-1"` is removed from the "cell" component and the cell is focused, sort cannot be triggered via keyboard as only a `click` event triggers sort.

**Comment by dtassone** [comment]
Please hold on this one. I want to build an easy way to map keys, and potentially do some refactoring in useKeyboard, as part of edit cells... 🤔

**Comment by oliviertassinari** [comment]
I would also argue that the issue has zero upvotes (outside of core team members). We could ignore it

**Comment by msteitle** [comment]
Thank you for tackling this a11y problem, @DanailH . It's greatly appreciated.

**Comment by metetunca** [comment]
It looks like, we can focus on headers with keyboard now, however sort direction indicator still doesn't show up when it is focused with keyboard. Is there way to solve this problem now?
